api/src/main/java/org/openmrs/Order.java

```diff
@@ -32,10 +32,18 @@
  * 
  * @version 1.0
  */
-public class Order extends BaseOpenmrsData implements java.io.Serializable {
+public class Order extends BaseOpenmrsData implements java.io.Serializable, Comparable{
```

Comment:
In general, orders should not be compared by sequence numbers outside the context of an OrderGroup, so you should have a separate comparator class, then you pass as instance of it to the TreeSet constructor in OrderGroup.getOrders. Imagine a user calls OrderService.getOrders(), then they sort them, not everyone wants them sorted by sequence number since they are not necessarily in OrderGroups